src/modules/ErrorHandler.js

```diff
@@ -0,0 +1,53 @@
+import store from '../state/store/configureStore';
+
+const errorHandler = (error) => {
+  if (error.message !== 'Network Error') {
+    switch (error.response.status) {
+      case 500:
+        store.dispatch({
+          type: 'ERROR_MESSAGE',
+          payload:
+            'Servers are currently not responding, Please try again later',
+        });
+        break;
+      case 400:
+        store.dispatch({
+          type: 'ERROR_MESSAGE',
+          payload: "Payment didn't go through, please try again.",
+        });
+        break;
+      case 401:
+        store.dispatch({
+          type: 'ERROR_MESSAGE',
+          payload:
+            'You are not authorised to do this, contact your system adminstrator',
+        });
+        break;
+      case 422:
+        store.dispatch({
+          type: 'ERROR_MESSAGE',
+          payload:
+            'Server is unable to process your request, please try again.',
+        });
+        break;
+      case 404:
+        store.dispatch({
+          type: 'ERROR_MESSAGE',
+          payload: error.response.data.error_message,
+        });
+        break;
+      default:
+        store.dispatch({
+          type: 'ERROR_MESSAGE',
+          payload: error.message,
+        });
+        break;
+    }
+  } else {
+    store.dispatch({
+      type: 'ERROR_MESSAGE',
+      payload: error.message,
+    });
+  }
```

Comment:
is it needed to have to of the same or am i just lost in this error jungle? ;)

Reply:
the if statment is checking for a network error, if this is true, we have the Error message show for that. Otherwise it continues in to the If statement and there we need a default answer if none of the cases are hit.